include/swift/SIL/OwnershipUtils.h

```diff
@@ -546,14 +564,40 @@ struct InteriorPointerOperand {
   InteriorPointerOperand(Operand *op)
       : operand(op), kind(*InteriorPointerOperandKind::get(op)) {}
 
-  /// If value is a borrow introducer return it after doing some checks.
+  /// If \p op has a user that is an interior pointer, return a valid
+  /// value. Otherwise, return None.
   static Optional<InteriorPointerOperand> get(Operand *op) {
     auto kind = InteriorPointerOperandKind::get(op);
     if (!kind)
       return None;
     return InteriorPointerOperand(op, *kind);
   }
 
+  /// If \p val is a result of an instruction that is an interior pointer,
+  /// return an interor pointer operand based off of the base value operand of
+  /// the instruction.
+  static Optional<InteriorPointerOperand>
```

Comment:
I dislike passing around and unwrapping optionals when the abstraction can have an invalid representation.

Reply:
Ok. I see your point. It would be great if c++ would let us do guard let like in swift = ). That would make this not an issue.